google/cloud/internal/big_endian_test.cc

```diff
@@ -13,67 +13,179 @@
 // limitations under the License.
 
 #include "google/cloud/internal/big_endian.h"
-#include <gmock/gmock.h>
-#include <cstring>
+#include "google/cloud/testing_util/assert_ok.h"
+#include <ios>
+#include <limits>
+#include <string>
+#include <vector>
 
 namespace google {
 namespace cloud {
 inline namespace GOOGLE_CLOUD_CPP_NS {
 namespace internal {
 namespace {
 
-TEST(BigEndianTest, Int16) {
-  std::uint8_t buf[] = {0x01, 0x02};
-  std::int16_t value;
-  static_assert(sizeof(value) == sizeof(buf), "Mismatched sizes");
-  std::memcpy(&value, buf, sizeof(buf));
-  EXPECT_EQ(0x0102, FromBigEndian(value));
-  EXPECT_EQ(value, FromBigEndian(ToBigEndian(value)));
+template <typename T>
+struct TestData {
+  T n;
+  std::string s;
+};
+
+template <typename T>
+void RunTests(std::vector<TestData<T>> const& test_data) {
+  for (auto const& td : test_data) {
+    std::string const encoded = EncodeBigEndian(td.n);
+    // The std::hex io manipulator doesn't work right with `char`, so we add 0
+    // (an int) to it so that small types are promoted to at least an int.
+    auto const for_hex = td.n + 0;
+    EXPECT_EQ(td.s, encoded) << "n=" << td.n << " hex=" << std::hex << for_hex;
+    auto const decoded = DecodeBigEndian<T>(encoded);
+    EXPECT_STATUS_OK(decoded) << "n=" << td.n << " hex=" << std::hex << for_hex;
+    EXPECT_EQ(td.n, *decoded);
+  }
+}
+
+TEST(RoundTripBigEndian, Int8) {
+  std::vector<TestData<std::int8_t>> test_data = {
+      {std::numeric_limits<std::int8_t>::min(), std::string("\x80", 1)},
```

Comment:
I'd replace all the numeric_limits::min/max() pieces with the actual expected values ... we're not testing numeric_limits.

Reply:
OK. Done.